cpp/ql/lib/semmle/code/cpp/ir/dataflow/internal/DataFlowPrivate.qll

```diff
@@ -729,25 +818,31 @@ predicate jumpStep(Node n1, Node n2) {
  * value of `node1`.
  *
  * The boolean `certain` is true if the destination address does not involve
- * any pointer arithmetic, and false otherwise.
+ * any pointer arithmetic, and false otherwise. This has to do with whether a
+ * store step can be used to clear a field (see `clearsContent`).
  */
-predicate storeStepImpl(Node node1, Content c, PostFieldUpdateNode node2, boolean certain) {
+predicate storeStepImpl(Node node1, Content c, Node node2, boolean certain) {
   exists(int indirectionIndex1, int numberOfLoads, StoreInstruction store |
     nodeHasInstruction(node1, store, pragma[only_bind_into](indirectionIndex1)) and
-    node2.getIndirectionIndex() = 1 and
-    numberOfLoadsFromOperand(node2.getFieldAddress(), store.getDestinationAddressOperand(),
-      numberOfLoads, certain)
+    node2.(PostFieldUpdateNode).getIndirectionIndex() = 1 and
```

Comment:
How about we store the value of `node2.(PostFieldUpdateNode)` in a local variable of type `PostFieldUpdateNode` to avoid this repeated casting?

Reply:
I don't think this code has actually been changed by the PR, aside from indenting.  But done.

Reply:
[It did](https://github.com/github/codeql/pull/15371/commits/8c0f02ac4b841cdfa159ff55a9fc0695e6e7af31#diff-314268058d269a10f8f014a41344742dde464d015809e4b3c32b560fa3a6e9a6R812), but I don't blame you for getting lost in the git history at this point 🙂. Thanks for fixing it!